### PR TITLE
uidCopy and uidStore

### PR DESCRIPTION
**Pull request by joehealy**
Not so confident with git, but wanting to also PR the addition of uidCopy and uidStore

### CODE REVIEW

**Comment by mkawalec** [comment]
sorry, life go in the way, I'll have a moment to review this and the other PR this weekend. Thank you a lot for that contribution!

**Comment by joehealy** [comment]
Any chance you can have a look at this PR? It would be great to get the uidCopy/uidStore functions in.

File: src/Network/IMAP.hs [hunk 1]
```diff
@@ -317,13 +319,28 @@ store conn sequenceSet dataItem flagList = do
                                      encodeUtf8 dataItem, flagsToText flagList]
   sendCommand conn command
 
+uidStore :: (MonadPlus m, MonadIO m, Universe m) => IMAPConnection ->
+  T.Text -> T.Text -> [Flag] -> m CommandResult
+uidStore conn sequenceSet dataItem flagList = do
+  let command = BSC.intercalate " " ["UID STORE", encodeUtf8 sequenceSet,
+                                     encodeUtf8 dataItem, flagsToText flagList]
+  liftIO $ putStrLn $ show command
```
Comment (mkawalec) [inline src/Network/IMAP.hs]: we don't need that

File: src/Network/IMAP/Parsers/Untagged.hs [hunk 2]
```diff
@@ -161,7 +161,7 @@ parseExpunge = do
 
 parseSearchResult :: Parser (Either ErrorMessage UntaggedResult)
 parseSearchResult = do
-  string "SEARCH "
-  msgIds <- AP.takeWhile1 isDigit `sepBy` char ' '
-  let parsedIds = mapM toInt msgIds
+  string "SEARCH"
+  msgIds <- AP.takeWhile isDigit `sepBy` char ' '
```
Comment (mkawalec) [inline src/Network/IMAP/Parsers/Untagged.hs]: is there a case when this caused a bug?

File: test/Network/IMAP/Tests.hs [hunk 3]
```diff
@@ -85,6 +85,19 @@ testRecent = do
   (res, _) <- runFakeIOWithReply conn "* 15 RECENT" "OK k" $ sendCommand conn "test"
   forUntagged res isRecent $ \(Recent count) -> count @?= 15
 
+
+testSearch1 = do
+  conn <- getConn
+  (res, _) <- runFakeIOWithReply conn "* SEARCH 583551 3" "OK UID SEARCH completed" $ sendCommand conn "test"
+  putStrLn $ show res
```
Comment (mkawalec) [inline test/Network/IMAP/Tests.hs]: no need for the output :)

File: src/Network/IMAP.hs (hunk 1 above)
Comment (joehealy) [reply src/Network/IMAP.hs]: I agree - sorry about that

File: src/Network/IMAP/Parsers/Untagged.hs (hunk 2 above)
Comment (joehealy) [reply src/Network/IMAP/Parsers/Untagged.hs]: I believe if there are no search results the original fails.

File: src/Network/IMAP.hs (hunk 1 above)
Comment (mkawalec) [reply src/Network/IMAP.hs]: no worries, just remove it and I'll merge the PR :)

File: src/Network/IMAP/Parsers/Untagged.hs (hunk 2 above)
Comment (mkawalec) [reply src/Network/IMAP/Parsers/Untagged.hs]: ah, good catch, thanks :)

**Comment by joehealy** [comment]
Finally removed that extraneous output. Are you able to merge (and ideally do a new release), or is there more I need to do?

Cheers,

Joe

**Comment by mkawalec** [comment]
@joehealy published it as 0.3.0.6

**Comment by joehealy** [comment]
great thanks

**Pull request merged**